enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,964 @@
+
+```bash
+$ opm index inspect --index=quay.io/some-namespace/my-index 
+$ // edit a my-index/package-config.json 
+$ opm index update my-index --tag=docker.io/some-namespace/my-index:latest
+```
+
+When an index is "inspected" for it's config files, a `image_sha256.json` will also be created that will store the sha of the index image being inspected. 
+```bash 
+$ cat community-operators/image_sha256.json
+{
+    "sha256":"9e0c9c2c8e973c7e17439509d1dc1512afa3a8ed7c0b169016299a357bbc5b10"
+}
+```
+When an update is attempted on an index image, the `update` command will verify that the sha information locally matches with the sha for the remote index image before overriding the content of the container image with the latest changes. This will prevent overriding of existing information that may not have been pulled locally due to asynchronous update of the config files. 
+
+> Note: For a seamless UX, the following things should be consider during implementation when the `opm index update` command is summoned: 
+> 1. Validate the files inside the folder given as input to the command to ensure they are valid json representation of packages (and not random files with random content, which could be potential security risks)
+> 2. If the files are valid files, verify that the sha information stored locally matches the sha of the container image from the registry it's hosted on. If they do not match, throw an error. 
+> 3. If the files are valid, and the sha is valid, take a git diff between the present content of the index (i.e the existing json representations) and the content of the folder given as input. Display the diff as an output of the command with the following message 
+> "The following changes will be updated in the index:"
+> If there is no diff between the content in the container image and the content of the local folder, throw an error with the message "No diff found between index and local content to update."    
+
+
+#### Creating new indexes using the config files from an existing index
+
+A new sub-command `create` will be introduced under `opm index`, that will take a folder containing json/yaml representation of packages, and a container image tag as input, and will create a new index with the package representations fed to it. 
+
+```bash
+$ tree community-operators 
+community-operators
+├── amqstreams.json
+└── etcd.json
+$ opm index create --from=community-operators --tag=docker.io/some-namespace/community-operators
+```
+
+Since `opm index add` will be modified to create the package config when bundles are added to the index, the `create` sub-command will first iterate over the local package configs and invoke `opm index add` for each bundle in the configs, and then replace the configs in the index image with the local configs so that the changes made to the configs locally are reflected in the remote index image.    
```

Comment:
This is confusing to me. Shouldn't `create` just append the json it finds, stick it in an image, and call it done? Or is this paragraph about maintaining the sql database in parallel to the json files?

Reply:
Remnants from before we decided to get rid of the db. Thanks for catching it I've removed this paragraph.